File: direct-file/backend/src/main/java/gov/irs/directfile/api/factgraph/FactDictionaryConfig.java
package gov.irs.directfile.api.factgraph;

import scala.collection.Iterable;
import scala.jdk.CollectionConverters;

public class FactDictionaryConfig implements gov.irs.factgraph.definitions.FactDictionaryConfigTrait {

    public FactDictionaryConfig(
            java.lang.Iterable<gov.irs.factgraph.definitions.fact.FactConfigTrait> facts,
            gov.irs.factgraph.definitions.meta.MetaConfigTrait meta) {
        this.facts = facts;
        this.meta = meta;
    }

    public java.lang.Iterable<gov.irs.factgraph.definitions.fact.FactConfigTrait> facts;
    public gov.irs.factgraph.definitions.meta.MetaConfigTrait meta;

    @Override
    public Iterable<gov.irs.factgraph.definitions.fact.FactConfigTrait> facts() {
        return CollectionConverters.IterableHasAsScala(facts).asScala().toSeq();
    }

    @Override
    public gov.irs.factgraph.definitions.meta.MetaConfigTrait meta() {
        return meta;
    }
}

File: direct-file/fact-graph-scala/js/src/main/scala/gov/irs/factgraph/definitions/FactDictionaryConfig.scala
package gov.irs.factgraph.definitions

import gov.irs.factgraph.definitions.fact.FactConfigElement
import gov.irs.factgraph.definitions.meta.MetaConfigTrait

import scala.scalajs.js.annotation.{JSExport, JSExportTopLevel}
import scala.scalajs.js
@JSExportTopLevel("FactDictionaryConfig")
object FactDictionaryConfig:
  @JSExport
  def create(
      meta: MetaConfigTrait,
      facts: scala.scalajs.js.Array[FactConfigElement],
  ): FactDictionaryConfigElement =
    new FactDictionaryConfigElement(meta, facts.toSeq)
